Source/santasyncservice/SNTSyncManager.m

```diff
@@ -45,7 +44,8 @@ @interface SNTSyncManager () {
 @property(nonatomic) dispatch_source_t fullSyncTimer;
 @property(nonatomic) dispatch_source_t ruleSyncTimer;
 
-@property(nonatomic) NSCache *dispatchLock;
+// Make sure to always lock and unlock from the same thread.
+@property(nonatomic) NSLock *syncLock;
```

Comment:
I found following the state of the `syncLock` to be quite difficult. There are some places which have some potential red flags, but it's hard to know if there is really an issue. For example in `ruleSyncSecondsFromNow`, this does a `tryLock` to check the current state of the lock, but then leaves the lock unlocked before returning to the caller. Such a point-in-time check doesn't provide much of a guarantee to callers and could easily lead to various race conditions.

It seems like using a dispatch queue onto which relevant chunks of the sync code could be submitted would likely increase readability and provide stronger state guarantees.